### ISSUE TITLE
TS1057: Type null is not an awaitable return type

### ISSUE THREAD

**Issue by lukas-zech-software** [Microsoft/TypeScript#14031]
**TypeScript Version:**  2.1.6

**Code**

```ts
const MISSING_DATA: null = null;

export class Test {
  public async get(): Promise<number|null> {
    const result = await Promise.resolve(null);
    if (result === null) {

      //error TS1059: Return expression in async function does not have a valid callable 'then' member.
      return MISSING_DATA;

      //error TS1059: Return expression in async function does not have a valid callable 'then' member.
      return Promise.resolve<null>(MISSING_DATA);

      // compiles
      return Promise.resolve<number|null>(MISSING_DATA);
    }

    return Promise.resolve(1);
  }
}
```

**Expected behavior:**
Code compiles

**Actual behavior:**
Error `error TS1059: Return expression in async function does not have a valid callable 'then' member.` is thrown when returning a variable of type `null`

We use a constant value to indicate missing/invalid values on our data. 
This constants is described with `const MISSING_DATA: null = null;`

When trying to return this value in a async method with the return type `Promise<number|null>` the mentioned error appears.

Why is `null` not an accepted return type for promises/async functions?

**Comment by lukas-zech-software** [comment]
Could be related to https://github.com/Microsoft/TypeScript/issues/5911#issuecomment-161736662 but my problems seems to be specific to the `null` type used as generic parameter for `Promise` and not with `Promise` itself

**Comment by rbuckton** [comment]
This does not repro in 2.3. It looks like it may have been fixed by #13134.
